### PR TITLE
Moved navigation bar to top + incorporated top nav bar design + azure profile link

### PR DESCRIPTION
**Pull request by annielovesjs**
Changes:
- moved navigation bar to top and made it fully responsive + functional 
- added additional styling to make the transition smoother 
- moved and rerouted the azure profile / login button to cloud services page (will adjust design more in future PR according to Christina's designs)
- changed some text

How to test
- check if the azure login button functions normally on cloud services page 
- UI should look like this: 
finishedHeader

### CODE REVIEW

File: src/client/src/components/TopNavBar/index.tsx [hunk 1]
```diff
@@ -15,27 +15,27 @@ import { isValidNameAndProjectPathSelector } from "../../selectors/wizardSelecti
...
   },
   summary: {
-    id: "leftSidebar.summary",
-    defaultMessage: "5. Summary"
+    id: "TopNavBar.summary",
+    defaultMessage: "Summary"
   },
   ariaNavLabel: {
-    id: "leftSideBar.ariaNavLabel",
+    id: "TopNavBar.ariaNavLabel",
```
Comment (jsondoo) [inline src/client/src/components/TopNavBar/index.tsx]: first of all good work 
we should do `topNavBar` for consistency

File: src/client/src/components/TopNavBar/index.tsx [hunk 2]
```diff
@@ -47,9 +47,9 @@ interface IStateProps {
 
 type Props = RouteComponentProps & IStateProps & InjectedIntlProps;
 
-const LeftSidebar = (props: Props) => {
+const TopNavBar = (props: Props) => {
   const { formatMessage } = props.intl;
-  const leftSidebarData: string[] = [
+  const TopNavBarData: string[] = [
```
Comment (jsondoo) [inline src/client/src/components/TopNavBar/index.tsx]: also here `topNavBarData`

File: src/client/src/components/TopNavBar/index.tsx [hunk 3]
```diff
@@ -60,23 +60,33 @@ const LeftSidebar = (props: Props) => {
...
+    event: React.SyntheticEvent,
+    idx: number,
+    visited: boolean
+  ) => {
+    if (visited) {
+      setPathIndex(ROUTES_ARRAY.indexOf(ROUTES_ARRAY[idx]));
+    }
+  };
   React.useEffect(() => {
+    console.log(ROUTES_ARRAY.indexOf(pathname));
```
Comment (jsondoo) [inline src/client/src/components/TopNavBar/index.tsx]: delete this fam

File: src/client/src/components/TopNavBar/index.tsx [hunk 4]
```diff
@@ -60,23 +60,33 @@ const LeftSidebar = (props: Props) => {
...
+      {
         <nav
-          className={classnames(styles.leftView, styles.container)}
+          className={classnames(styles.topNavBar)}
           aria-label={intl.formatMessage(messages.ariaNavLabel)}
         >
           <div>
-            {leftSidebarData.map((sidebartitle, idx) => {
+            {TopNavBarData.map((sidebartitle, idx) => {
+              console.log("currentPathIndex " + currentPathIndex);
```
Comment (jsondoo) [inline src/client/src/components/TopNavBar/index.tsx]: some more console logs in this PR

File: src/client/src/containers/SelectFrameworks/index.tsx [hunk 5]
```diff
@@ -43,7 +43,7 @@ class SelectFrameworks extends React.Component<Props> {
 
   render() {
     return (
-      <div className={styles.container}>
```
Comment (jsondoo) [inline src/client/src/containers/SelectFrameworks/index.tsx:43]: just wondering, why was this removed?

File: src/client/src/mockData/leftSidebarData.ts [hunk 6]
```diff
@@ -1,10 +1,10 @@
...
   "2. Project Type",
   "3. Frameworks",
   "4. Pages",
-  "5. Services (Optional)",
+  "5. Add Optional Cloud Services",
   "6. Summary"
 ];
 
-export default leftSidebarData;
+export default TopNavBarData;
```
Comment (jsondoo) [inline src/client/src/mockData/leftSidebarData.ts]: i think we can rename this file to TopNavBarData

File: src/client/src/components/TopNavBar/index.tsx (hunk 3 above)
Comment (annielovesjs) [reply src/client/src/components/TopNavBar/index.tsx]: oopsies

File: src/client/src/containers/SelectFrameworks/index.tsx (hunk 5 above)
Comment (annielovesjs) [reply src/client/src/containers/SelectFrameworks/index.tsx:43]: it made the select frameworks page inconsistent with all the other pages - the only noticeable difference .container made was additional padding that made it look worse LOL

File: src/client/src/mockData/leftSidebarData.ts (hunk 6 above)
Comment (annielovesjs) [reply src/client/src/mockData/leftSidebarData.ts]: oh my bad - good catch!

File: src/client/src/containers/AzureLogin/index.tsx [hunk 7]
```diff
@@ -42,11 +46,38 @@ class AzureLogin extends React.Component<Props> {
...
     });
   };
+  signOutClick = () => {
+    this.props.vscode.postMessage({
+      module: EXTENSION_MODULES.AZURE,
+      command: EXTENSION_COMMANDS.AZURE_LOGOUT,
+      track: true
+    });
+  };
+  keyDownClick = (event: any) => {
```
Comment (ngkelly3) [inline src/client/src/containers/AzureLogin/index.tsx]: keydown or click event? You can type these events using https://stackoverflow.com/questions/38337193/import-event-types-from-react-typescript

the nav looks pretty nice!

Comment (annielovesjs) [reply src/client/src/containers/AzureLogin/index.tsx]: thanks for the tip! Just added it

File: src/client/src/containers/AzureLogin/index.tsx [hunk 8]
```diff
@@ -42,11 +46,38 @@ class AzureLogin extends React.Component<Props> {
...
+    if (event.keyCode === 13 || event.keyCode === 32) {
```
Comment (ngkelly3) [inline src/client/src/containers/AzureLogin/index.tsx]: Optional nit: Sometimes it's useful to leave a comment about what the keycodes represent so devs don't have to look them up.

Comment (SahilTara) [reply src/client/src/containers/AzureLogin/index.tsx]: Feel like these are going to be replaced by .key events so i think its okay

File: src/client/src/containers/SelectFrameworks/styles.module.css [hunk 9]
```diff
@@ -1,4 +1,4 @@
-.container {
+/* .container {
```
Comment (SahilTara) [inline src/client/src/containers/SelectFrameworks/styles.module.css]: Why are we commenting this out, do we still need it?

File: src/client/src/containers/AzureLogin/index.tsx (hunk 8 above)
Comment (annielovesjs) [reply src/client/src/containers/AzureLogin/index.tsx]: Aside from my file changes, there's a lot of areas in the codebase where the format is similar so I think instead of adding comments for just one of them, we can replace all event.keyCode with eventkey (like sahil said) and check for the type of key pressed as a future improvement in a different pr

**Comment by SahilTara** [comment]
This is the wizard in light theme:
image

feel like theres a readability issue here

File: src/client/src/components/TopNavBar/index.tsx (hunk 1 above)
Comment (annielovesjs) [reply src/client/src/components/TopNavBar/index.tsx]: done :)

File: src/client/src/containers/SelectFrameworks/styles.module.css (hunk 9 above)
Comment (annielovesjs) [reply src/client/src/containers/SelectFrameworks/styles.module.css]: I forgot to remove this - we don't need it anymore

**Comment by Tanya0609** [comment]
There are few cases the top nav is not matching the actual page. Some investigations will be needed.
not-match

File: src/client/src/containers/AzureLogin/index.tsx (hunk 8 above)
Comment (jsondoo) [reply src/client/src/containers/AzureLogin/index.tsx]: @SahilTara recommended this before so we have an issue for this #817

Comment (ngkelly3) [reply src/client/src/containers/AzureLogin/index.tsx]: Good to know! Could help a future dev change the code a bit more easily? 🙃

Comment (SahilTara) [reply src/client/src/containers/AzureLogin/index.tsx]: Honestly, after some thought, I don't think its a good idea to do it the wrong way just because its being done like that everywhere else. I think that if you keep it as keyCode here, it will just add to code debt. Why do it right later when you can do it properly right now?

Comment (annielovesjs) [reply src/client/src/containers/AzureLogin/index.tsx]: @SahilTara I'm thinking of doing it in a separate PR just for the issue because currently there are about 100 files that would need to change based on that. It would make this PR a lot more difficult to review if I do it here

**Comment by annielovesjs** [comment]
@Tanya0609 I resolved the issue in a new commit :)

**Comment by annielovesjs** [comment]
> There are few cases the top nav is not matching the actual page. Some investigations will be needed.
> not-match

@Tanya0609 I resolved the issue in a new commit :)

**Comment by annielovesjs** [comment]
> This is the wizard in light theme:
> image
> 
> feel like theres a readability issue here

That's a super good catch - I forgot to account for light mode when making the styling changes. I made some changes to improve its readability and it looks like this now: 
lightMode

File: src/client/src/containers/AzureLogin/index.tsx (hunk 8 above)
Comment (SahilTara) [reply src/client/src/containers/AzureLogin/index.tsx]: No no I'm not saying change it for everywhere in the code base (that would be an unreasonable request), i'm saying for the new code that you're adding. It'll be one less file you'll have to do later.

**Comment by SahilTara** [comment]
> > This is the wizard in light theme:
> > image
> > feel like theres a readability issue here
> 
> That's a super good catch - I forgot to account for light mode when making the styling changes. I made some changes to improve its readability and it looks like this now:
> lightMode

@annielovesjs It feels a bit different from the dark theme version, no outline/circle on the non selected ones. But yeah that really helps with the readability.

File: src/client/src/containers/AzureLogin/index.tsx (hunk 8 above)
Comment (annielovesjs) [reply src/client/src/containers/AzureLogin/index.tsx]: Ohhh I see! haha okay I'll make the change then

**Comment by annielovesjs** [comment]
> > > This is the wizard in light theme:
> > > image
> > > feel like theres a readability issue here
> > 
> > 
> > That's a super good catch - I forgot to account for light mode when making the styling changes. I made some changes to improve its readability and it looks like this now:
> > lightMode
> 
> @annielovesjs It feels a bit different from the dark theme version, no outline/circle on the non selected ones. But yeah that really helps with the readability.

I got you fam: 
lightmodenavbar

**Review by jasmingoh — DISMISSED** [review]

**Review by jcoc611-microsoft — APPROVED** [review]

**Comment by jasmingoh** [comment]
@annielovesjs,  just want to make sure, should the cursor be a pointer for pages that have not been visited?

**Comment by annielovesjs** [comment]
> @annielovesjs, just want to make sure, should the cursor be a pointer for pages that have not been visited?

its actually not pointer for the unvisited pages - only for the selected and visited pages

**Review by jasmingoh — APPROVED** [review]

**Review by jsondoo — APPROVED** [review]
ship it
